### PR TITLE
🐛Source Amazon Ads: fix `portfolioId` type

### LINKED ISSUE #17623: Source Amazon Ads: schema typed incorrectly, "Could not evaluate union" exception
## Environment
- **Airbyte version**:  0.40.10
- **OS Version / Instance**:  Ubuntu 22.04, GCP
- **Deployment**: Docker 
- **Source Connector and version**: Amazon Ads 0.1.22
- **Destination Connector and version**: BigQuery (denormalized typed struct) 1.2.4
- **Step where error happened**: BigQuery (denormalized typed struct) Destination connector on time sync

## Current Behavior
I am regularly getting errors like this.

```
2022-09-16 23:07:15 destination > Exception while accepting message
Stack Trace: tech.allegro.schema.json2avro.converter.AvroConversionException: Failed to convert JSON to Avro: Could not evaluate union, field portfolioId is expected to be one of these: NULL, STRING. If this is a complex type, check if offending field (path: portfolioId) adheres to schema: 130457359476581
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.read(JsonGenericRecordReader.java:129)
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.read(JsonGenericRecordReader.java:118)
	at tech.allegro.schema.json2avro.converter.JsonAvroConverter.convertToGenericDataRecord(JsonAvroConverter.java:95)
	at io.airbyte.integrations.destination.s3.avro.AvroRecordFactory.getAvroRecord(AvroRecordFactory.java:44)
	at io.airbyte.integrations.destination.bigquery.BigQueryAvroSerializedBuffer.writeRecord(BigQueryAvroSerializedBuffer.java:46)
	at io.airbyte.integrations.destination.record_buffer.BaseSerializedBuffer.accept(BaseSerializedBuffer.java:92)
	at io.airbyte.integrations.destination.record_buffer.SerializedBufferingStrategy.addRecord(SerializedBufferingStrategy.java:61)
	at io.airbyte.integrations.destination.buffered_stream_consumer.BufferedStreamConsumer.acceptTracked(BufferedStreamConsumer.java:138)
	at io.airbyte.integrations.base.FailureTrackingAirbyteMessageConsumer.accept(FailureTrackingAirbyteMessageConsumer.java:47)
	at io.airbyte.integrations.base.IntegrationRunner.consumeMessage(IntegrationRunner.java:259)
	at io.airbyte.integrations.base.IntegrationRunner.consumeWriteStream(IntegrationRunner.java:172)
	at io.airbyte.integrations.base.IntegrationRunner.lambda$runConsumer$2(IntegrationRunner.java:178)
	at io.airbyte.integrations.base.IntegrationRunner.watchForOrphanThreads(IntegrationRunner.java:207)
	at io.airbyte.integrations.base.IntegrationRunner.runConsumer(IntegrationRunner.java:177)
	at io.airbyte.integrations.base.IntegrationRunner.runInternal(IntegrationRunner.java:146)
	at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:97)
	at io.airbyte.integrations.destination.bigquery.BigQueryDenormalizedDestination.main(BigQueryDenormalizedDestination.java:200)
Caused by: org.apache.avro.AvroTypeException: Could not evaluate union, field portfolioId is expected to be one of these: NULL, STRING. If this is a complex type, check if offending field (path: portfolioId) adheres to schema: 130457359476581
	at tech.allegro.schema.json2avro.converter.AvroTypeExceptions.unionException(AvroTypeExceptions.java:28)
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.readUnion(JsonGenericRecordReader.java:294)
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.read(JsonGenericRecordReader.java:196)
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.read(JsonGenericRecordReader.java:170)
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.lambda$readRecord$0(JsonGenericRecordReader.java:152)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.readRecord(JsonGenericRecordReader.java:141)
	at tech.allegro.schema.json2avro.converter.JsonGenericRecordReader.read(JsonGenericRecordReader.java:127)
	... 16 more
```

The full log is attached. 
[logs-10.txt](https://github.com/airbytehq/airbyte/files/9590510/logs-10.txt)

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
*Fix `portfolioId` type to integer ([Docs](https://advertising.amazon.com/API/docs/en-us/reporting/v3/metrics#portfolioid))*

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-amazon-ads/Dockerfile [hunk 1]
```diff
@@ -14,5 +14,5 @@ ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
 
 
-LABEL io.airbyte.version=1.0.4
+LABEL io.airbyte.version=1.0.5
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-amazon-ads/Dockerfile:17]:
```suggestion
LABEL io.airbyte.version=2.0.0
```

File: airbyte-integrations/connectors/source-amazon-ads/metadata.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: c6b0a29e-1da9-4512-9002-7bfd0cba2246
-  dockerImageTag: 1.0.4
+  dockerImageTag: 1.0.5
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-amazon-ads/metadata.yaml:11]:
```suggestion
  dockerImageTag: 2.0.0
```

File: docs/integrations/sources/amazon-ads.md [hunk 3]
```diff
@@ -94,7 +94,8 @@ Information about expected report generation waiting time you may find [here](ht
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                         |
 |:--------|:-----------|:---------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------|
-| 1.0.4   | 2023-05-04 | [00000](https://github.com/airbytehq/airbyte/pull/00000) | Add availability strategy for basic streams (not including report streams)                                      |
+| 1.0.5   | 2023-05-08 | [25874](https://github.com/airbytehq/airbyte/pull/25874) | Type `portfolioId` as integer                                                                                   |
```
Comment (alafanechere) [inline docs/integrations/sources/amazon-ads.md:97]:
```suggestion
| 2.0.0   | 2023-05-08 | [25874](https://github.com/airbytehq/airbyte/pull/25874) | Type `portfolioId` as integer                                                                                   |
```

**Review by alafanechere — CHANGES_REQUESTED** [review]
@roman-yermilov-gl this type change is a breaking change. I'm going to create a breaking change release plan that I will share in this PR. Could you please link the original issue that led you to implement this change? Is there any functional implication in keeping these fields as decimal today? If it prevents you from deserializing the API response I would suggest to use a type transformer. If this is leading to unusable data on the destination side this change is indeed required.

**Comment by alafanechere** [comment]
@roman-yermilov-gl here's the [breaking change release plan draft](https://docs.google.com/document/d/1zWtBOmn85IqxecgXpNHY_72T3ZubsGGGUN8skxByH2c/edit#). I'd need more context on why this change is introduced to move forward.

**Review by alafanechere — CHANGES_REQUESTED** [review]
@roman-yermilov-gl do you confirm the modified streams are present in the configured catalog we use for acceptance tests? Could you please update the expected records file with data corresponding to the modified streams?

**Comment by alafanechere** [comment]
@roman-yermilov-gl thank you for the details. I updated the breaking change document. I suspect the error mentioned #17623 to only impact BigQuery denormalized. As we have no test data it's hard to assess the breadth of this problem. I suggest to not merge it until we either:
* Have more evidence that this bug is widespread and impacting multiple users
* Having a painless release strategy for breaking change

**Comment by roman-yermilov-gl** [comment]
> @roman-yermilov-gl thank you for the details. I updated the breaking change document. I suspect the error mentioned #17623 to only impact BigQuery denormalized. As we have no test data it's hard to assess the breadth of this problem. I suggest to not merge it until we either:
> 
> * Have more evidence that this bug is widespread and impacting multiple users
> * Having a painless release strategy for breaking change

I agree but what would be your suggestion for next steps? Should I put the task https://github.com/airbytehq/airbyte/issues/17623 on hold?

**Comment by alafanechere** [comment]
> I agree but what would be your suggestion for next steps? Should I put the task https://github.com/airbytehq/airbyte/issues/17623 on hold? 

Yes I think so. I'll ask on the issue to try to seed the account with seed data.

**Comment by roman-yermilov-gl** [comment]
> > I agree but what would be your suggestion for next steps? Should I put the task #17623 on hold?
> 
> Yes I think so. I'll ask on the issue to try to seed the account with seed data.

Hi @alafanechere 
Is there updates on this isssue?

**Comment by alafanechere** [comment]
@roman-yermilov-gl I got not update from the stakeholders on #17623
Now that [we can feature flag connectors](https://airbytehq-team.slack.com/archives/C02UF50V9HA/p1685026337296189) on cloud I encourage you to take ownership on this [doc](https://docs.google.com/document/d/1zWtBOmn85IqxecgXpNHY_72T3ZubsGGGUN8skxByH2c/edit#heading=h.r2xzw0kclodc) and suggest a rollout plan in coordination with TCS.

**Comment by roman-yermilov-gl** [comment]
/publish connector=connectors/source-amazon-ads pre-release=true run-tests=false

**Comment by alafanechere** [comment]
Updating the branch to get a fresh CI run.

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`fa1f4176a2`](https://github.com/airbytehq/airbyte/pull/25874/commits/fa1f4176a25528d07e473c7299ca6fdaebd762aa)) - ❌

⏲️  Total pipeline duration: 197 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ❌        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5279427071)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`39f549b39c`](https://github.com/airbytehq/airbyte/pull/25874/commits/39f549b39cc0f052861afff637429cba9588c819)) - ❌

⏲️  Total pipeline duration: 195 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ❌        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5322952165)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`9c41aeb7e6`](https://github.com/airbytehq/airbyte/pull/25874/commits/9c41aeb7e6c08ae3dd6a4f8208285849ec3cb4eb)) - ✅

⏲️  Total pipeline duration: 194 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5324540529)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Review by alafanechere — APPROVED** [review]

**Review by bnchrch — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`221cdc4104`](https://github.com/airbytehq/airbyte/pull/25874/commits/221cdc4104092baecfa47fcc5a4d784507139021)) - ✅

⏲️  Total pipeline duration: 211 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5324979540)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Pull request merged**